From d56ce647b3d423ec0e2d7a1e63ec4efb21072c45 Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@st.com>
Date: Thu, 16 Dec 2021 15:37:53 +0100
Subject: [PATCH 040/841] feat(stm32mp1): save zq0cr0 in backup context at boot
 time

Save the zq0cr0 zdata value at boot time. It will avoid
to do that later in the critical section as the value
will remain the same.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 44ccec909deff35b13bf5f933984f8b58aff4b91 feat(stm32mp1): save zq0cr0 in backup context at boot time]
Signed-off-by: Lionel Debieve <lionel.debieve@foss.st.com>
Change-Id: Id6d80e5156b27e9709564ed16851efc20824cc29
---
 plat/st/stm32mp1/stm32mp1_context.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/plat/st/stm32mp1/stm32mp1_context.c b/plat/st/stm32mp1/stm32mp1_context.c
index f1109bd48..ff1022924 100644
--- a/plat/st/stm32mp1/stm32mp1_context.c
+++ b/plat/st/stm32mp1/stm32mp1_context.c
@@ -9,6 +9,7 @@
 #include <arch_helpers.h>
 #include <common/debug.h>
 #include <drivers/clk.h>
+#include <drivers/st/stm32mp1_ddr_helpers.h>
 #include <drivers/st/stm32mp1_ddr_regs.h>
 #include <dt-bindings/clock/stm32mp1-clks.h>
 #include <lib/utils.h>
@@ -121,6 +122,7 @@ void stm32_context_save_bl2_param(void)
 	backup_data->bl2_code_end = BL_CODE_END;
 	backup_data->bl2_end = BL2_END;
 	backup_data->magic = MAILBOX_MAGIC_V3;
+	backup_data->zq0cr0_zdata = ddr_get_io_calibration_val();
 
 	clk_disable(BKPSRAM);
 }
-- 
2.47.2

